xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -380,9 +382,6 @@ func (b *cdsBalancer) run() {
 			case *watchUpdate:
 				b.handleWatchUpdate(update)
 			}
-
-		// Close results in cancellation of the CDS watch and closing of the
-		// underlying edsBalancer and is the only way to exit this goroutine.
```

Comment:
This comment and the one below are still true.  Did you feel they were redundant?

Reply:
Initially I had moved stuff from here into `Close()`. At that point, they were clearly out of place. But that cause a problem with the race detector. So, I moved them back here and added the `done` event. At that point I didn't add it back because, yes, I felt they were a bit redundant.